optaplanner-core/src/main/java/org/optaplanner/core/api/score/stream/uni/UniConstraintStream.java

```diff
@@ -218,6 +218,88 @@
+     * @param otherClass never null
+     * @param joiner never null
+     * @param <B> the type of the second matched fact
+     * @return a stream that matches every A where B exists for which the {@link BiJoiner} is true
+     */
+    default <B> UniConstraintStream<A> ifExists(Class<B> otherClass, BiJoiner<A, B> joiner) {
+        return ifExists(otherClass, new BiJoiner[] { joiner });
+    }
+
+    /**
+     * As defined by {@link #ifExists(Class, BiJoiner)}.
+     * @param otherClass never null
+     * @param joiner1 never null
+     * @param joiner2 never null
+     * @param <B> the type of the second matched fact
+     * @return a stream that matches every A where B exists for which all the {@link BiJoiner}s are true
+     */
+    default <B> UniConstraintStream<A> ifExists(Class<B> otherClass, BiJoiner<A, B> joiner1, BiJoiner<A, B> joiner2) {
+        return ifExists(otherClass, new BiJoiner[] {joiner1, joiner2});
+    }
+
+    /**
+     * As defined by {@link #ifExists(Class, BiJoiner)}.
+     * @param otherClass never null
+     * @param joiner1 never null
+     * @param joiner2 never null
+     * @param joiner3 never null
+     * @param <B> the type of the second matched fact
+     * @return a stream that matches every A where B exists for which all the {@link BiJoiner}s are true
+     */
+    default <B> UniConstraintStream<A> ifExists(Class<B> otherClass, BiJoiner<A, B> joiner1, BiJoiner<A, B> joiner2,
+            BiJoiner<A, B> joiner3) {
+        return ifExists(otherClass, new BiJoiner[] {joiner1, joiner2, joiner3});
+    }
+
+    /**
+     * As defined by {@link #ifExists(Class, BiJoiner)}.
+     * @param otherClass never null
+     * @param joiner1 never null
+     * @param joiner2 never null
+     * @param joiner3 never null
+     * @param joiner4 never null
+     * @param <B> the type of the second matched fact
+     * @return a stream that matches every A where B exists for which all the {@link BiJoiner}s are true
+     */
+    default <B> UniConstraintStream<A> ifExists(Class<B> otherClass, BiJoiner<A, B> joiner1, BiJoiner<A, B> joiner2,
+            BiJoiner<A, B> joiner3, BiJoiner<A, B> joiner4) {
+        return ifExists(otherClass, new BiJoiner[] {joiner1, joiner2, joiner3, joiner4});
+    }
+
+    /**
+     * As defined by {@link #ifExists(Class, BiJoiner)}.
+     * <p>
+     * This method causes <i>Unchecked generics array creation for varargs parameter</i> warnings,
+     * but we can't fix it with a {@link SafeVarargs} annotation because it's an interface method.
+     * Therefore, there are overloaded methods with up to 4 {@link BiJoiner} parameters.
+     * @param otherClass never null
+     * @param joiners never null
+     * @param <B> the type of the second matched fact
+     * @return a stream that matches every A where B exists for which all the {@link BiJoiner}s are true
+     */
+    <B> UniConstraintStream<A> ifExists(Class<B> otherClass, BiJoiner<A, B>... joiners);
```

Comment:
Suggested solution: add a default `ifExists(Class<B> otherClass, BiJoiner<A,B>... joiners); that calls`ifExists(Class<B> otherClass, List<BiJoiner> joiners)`

```
default <B> UniConstraintStream<A> ifExists(Class<B> otherClass, BiJoiner<A, B>... joiners) {
    ifExists(otherClass, Arrays.asList(joiners));
}

<B>  UniConstraintStream<A> ifExists(Class<B> otherClass, List<BiJoiner<A, B>> joiners);

```

Reply:
Chris, I don't have anything against this change, but if we do it, we need to do the same for joins and perhaps in couple other places. All in all, that's a large amount of work, and scope creep in this context.

Would you mind filing a JIRA for this? You could even tackle it yourself, if you want to feel like diving into the CS code.

Reply:
I will first try it out in an example project to make sure my suggested change actually works as intended; if so, I will file a JIRA.

Reply:
Cannot make an interface method final, which is a blocker for `@SafeVarargs`. If you want to use this suggestion, we have two options:
1. Turn the interfaces into abstract classes (which allow us to use final).
2. Create a static wrapper.
I think both of those options are bad.

If you allow `@SuppressWarnings`, there a third option:
```
public interface TestInterface<A> {
    
    default <B> List<A> myMethodVarargs(Class<B> otherClass, @SuppressWarnings("unchecked") A... data) {
        return myMethodList(otherClass, Arrays.asList(data));
    }
    
    <B> List<A> myMethodList(Class<B> otherClass, List<A> data);
}
```

Reply:
Actually, thinking about this again, could you perhaps explain why you would need a method like that?

Personally I consider this:

   ifExists(Something.class, joiner1, joiner2, joiner3, joiner4)

To be much better than this:

   ifExists(Something.class, Arrays.asList(joiner1, joiner2, joiner3, joiner4);

But perhaps there is some aspect that I'm missing.

Reply:
It to make implementing the interface not inherit the varargs warnings (and working with Lists is easier than working with arrays). The varargs option is still available (under the same name; above was from testing trying to find a way to remove the warning). `@SafeVarargs` might work on implementing classes, as long as the methods are marked as `final`.

Reply:
I don't want to introduce methods that accepts List or Collection in this API.
Motivation:
- Don't proliferate more combinations of each method (join(List) vs join(vararg) and that for join, ifExists, joinOther, ... and then for Uni, Bi, Tri, ...). Not worth the maintenance burden I'd argue.
-  We're not really accepting an array - we're accepting a vararg. I don't see use cases at this point in time where users would actually use an array. I do see cases where they would provide 7+ parameters, hence the need for a vararg.